applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,540 @@
+	case SUB_STATE_SEARCH:
+		return "SUB_STATE_SEARCH";
+	default:
+		return "Unknown";
+	}
+}
+
+static void state_set(enum state_type new_state)
+{
+	if (new_state == state) {
+		LOG_DBG("State: %s", state2str(state));
+		return;
+	}
+
+	LOG_DBG("State transition %s --> %s",
+		state2str(state),
+		state2str(new_state));
+
+	state = new_state;
+}
+
+static void sub_state_set(enum sub_state_type new_state)
+{
+	if (new_state == sub_state) {
+		LOG_DBG("Sub state: %s", sub_state2str(sub_state));
+		return;
+	}
+
+	LOG_DBG("Sub state transition %s --> %s",
+		sub_state2str(sub_state),
+		sub_state2str(new_state));
+
+	sub_state = new_state;
+}
+
+/* Handlers */
+static bool app_event_handler(const struct app_event_header *aeh)
+{
+	if (is_app_module_event(aeh)) {
+		struct app_module_event *event = cast_app_module_event(aeh);
+		struct gnss_msg_data msg = {
+			.module.app = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_data_module_event(aeh)) {
+		struct data_module_event *event = cast_data_module_event(aeh);
+		struct gnss_msg_data msg = {
+			.module.data = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_util_module_event(aeh)) {
+		struct util_module_event *event = cast_util_module_event(aeh);
+		struct gnss_msg_data msg = {
+			.module.util = *event
+		};
+
+		message_handler(&msg);
+	}
+
+#if 0
+	if (is_location_module_event(aeh)) {
+		struct location_module_event *event = cast_location_module_event(aeh);
+		struct location_msg_data msg = {
+			.module.location = *event
+		};
+
+		message_handler(&msg);
+	}
+#endif
+	if (is_gnss_module_event(aeh)) {
+		struct gnss_module_event *event = cast_gnss_module_event(aeh);
+		struct gnss_msg_data msg = {
+			.module.gnss = *event
+		};
+
+		message_handler(&msg);
+	}
+
+	if (is_modem_module_event(aeh)) {
+		struct modem_module_event *event = cast_modem_module_event(aeh);
+		struct gnss_msg_data msg = {
+			.module.modem = *event
+		};
+
+		message_handler(&msg);
+	}
+	return false;
+}
+
+static void inactivity_timeout_handler(struct k_timer *timer_id)
+{
+	/* Simulate a sleep event from GNSS with older MFWs. */
+	int event = NRF_MODEM_GNSS_EVT_SLEEP_AFTER_TIMEOUT;
+
+	k_msgq_put(&event_msgq, &event, K_NO_WAIT);
+}
+
+static K_TIMER_DEFINE(inactivity_timer, inactivity_timeout_handler, NULL);
+
+static uint8_t set_satellites_tracked(struct nrf_modem_gnss_sv *sv, uint8_t sv_entries)
+{
+	uint8_t sv_used = 0;
+
+	/* Check number of tracked Satellites. SV number 0 indicates that the satellite was not
+	 * tracked.
+	 */
+	for (int i = 0; i < sv_entries; i++) {
+		if (sv[i].sv) {
+			sv_used++;
+		}
+	}
+
+	return sv_used;
+}
+
+static void timeout_send(void)
+{
+	struct gnss_module_event *gnss_module_event = new_gnss_module_event();
+
+	gnss_module_event->type = GNSS_EVT_TIMEOUT;
+
+	APP_EVENT_SUBMIT(gnss_module_event);
+}
+
+/* Static module functions. */
+static void data_send_pvt(void)
+{
+	struct gnss_module_event *gnss_module_event = new_gnss_module_event();
```

Comment:
I'm still using GNSS module events throughout location module because that was an easy way of getting everything working. If we have a longer term plan to remove GNSS module, I should use location module events (the file exists but not used anywhere yet), right? If we think we'll keep GNSS module too, I guess it's not clear whether to create new events for same functionality. I'm just about to start using location module events in location module but does this make sense?
@simensrostad Opinions?

Reply:
I don't quite see why location and GNSS module should coexist. Why not just convert it into the location module? Yes, that's typically how its done. Each module has its own header file with events that are sent **out** of the module.

Reply:
So should I remove GNSS module already now? I've kept it for comparison so you could use either GNSS or location module easily but you cannot enable both at the same time.

Reply:
I wouldn't mind. @jhn-nordic opinion?

Reply:
GNSS module removed now.